clowder/clowder_test.go

```diff
@@ -0,0 +1,207 @@
+	originalTopicName := "topic1"
+
+	brokerCfg := kafka.BrokerConfiguration{
+		Topic: originalTopicName,
+	}
+	kafkaTopics := map[string]api.TopicConfig{
+		"topic2": {
+			Name: "AnotherTopicName",
+		},
+	}
+
+	output, _ := capture.StandardOutput(func() {
+		clowder.UseClowderTopics(&brokerCfg, kafkaTopics)
+	})
+	assert.Equal(t, originalTopicName, brokerCfg.Topic, "topic name should not change")
+	assert.Contains(t, output, "warning: no kafka mapping found for topic topic1")
+}
+
+func TestUseBrokerConfigNoKafkaConfig(t *testing.T) {
+	brokerCfg := kafka.BrokerConfiguration{}
+	loadedConfig := api.AppConfig{}
+
+	output, _ := capture.StandardOutput(func() {
+		clowder.UseBrokerConfig(&brokerCfg, &loadedConfig)
+	})
+	assert.Contains(t, output, clowder.NoBrokerCfg)
+}
+
+func TestUseBrokerConfigNoKafkaBrokers(t *testing.T) {
+	brokerCfg := kafka.BrokerConfiguration{}
+	loadedConfig := api.AppConfig{
+		Kafka: &api.KafkaConfig{},
+	}
+
+	output, _ := capture.StandardOutput(func() {
+		clowder.UseBrokerConfig(&brokerCfg, &loadedConfig)
+	})
+	assert.Contains(t, output, clowder.NoBrokerCfg)
+}
+
+func TestUseBrokerConfigNoAuthNoPort(t *testing.T) {
+	brokerCfg := kafka.BrokerConfiguration{}
+	loadedConfig := api.AppConfig{
+		Kafka: &api.KafkaConfig{
+			Brokers: []api.BrokerConfig{
+				{
+					Hostname: "test_broker",
+					Port:     nil,
+				},
+			},
+		},
+	}
+
+	clowder.UseBrokerConfig(&brokerCfg, &loadedConfig)
+	assert.Equal(t, "test_broker", brokerCfg.Address)
+}
+
+func TestUseBrokerConfigNoAuth(t *testing.T) {
+	brokerCfg := kafka.BrokerConfiguration{}
+	port := 12345
+	loadedConfig := api.AppConfig{
+		Kafka: &api.KafkaConfig{
+			Brokers: []api.BrokerConfig{
+				{
+					Hostname: "test_broker",
+					Port:     &port,
+				},
+			},
+		},
+	}
+
+	clowder.UseBrokerConfig(&brokerCfg, &loadedConfig)
+	assert.Equal(t, "test_broker:12345", brokerCfg.Address)
+}
+
+func TestUseBrokerConfigAuthEnabledNoSasl(t *testing.T) {
+	brokerCfg := kafka.BrokerConfiguration{}
+	port := 12345
+	authType := api.BrokerConfigAuthtypeSasl
+	loadedConfig := api.AppConfig{
+		Kafka: &api.KafkaConfig{
+			Brokers: []api.BrokerConfig{
+				{
+					Hostname: "test_broker",
+					Port:     &port,
+					Authtype: &authType,
+				},
+			},
+		},
+	}
+
+	output, _ := capture.StandardOutput(func() {
+		clowder.UseBrokerConfig(&brokerCfg, &loadedConfig)
+	})
+
+	assert.Equal(t, "test_broker:12345", brokerCfg.Address)
+	assert.Contains(t, output, clowder.NoSaslCfg)
+}
+
+func TestUseBrokerConfigAuthEnabledWithSaslConfig(t *testing.T) {
+	brokerCfg := kafka.BrokerConfiguration{}
+	port := 12345
+	saslCfg := "user_pwd"
+	protocol := "tls"
+
+	authType := api.BrokerConfigAuthtypeSasl
+	loadedConfig := api.AppConfig{
+		Kafka: &api.KafkaConfig{
+			Brokers: []api.BrokerConfig{
+				{
+					Hostname: "test_broker",
+					Port:     &port,
+					Authtype: &authType,
+					Sasl: &api.KafkaSASLConfig{
+						Password:      &saslCfg,
```

Comment:
why using the same values for password, username and sasl mechanism? I would use a different one in order to make sure we are using the right fields